test/functional/multiwallet.py

```diff
@@ -15,30 +15,41 @@ class MultiWalletTest(BitcoinTestFramework):
     def set_test_params(self):
         self.setup_clean_chain = True
         self.num_nodes = 1
-        self.extra_args = [['-wallet=w1', '-wallet=w2', '-wallet=w3']]
+        self.extra_args = [['-wallet=w1.dat', '-wallet=w2.dat', '-wallet=w3.dat']]
 
     def run_test(self):
-        assert_equal(set(self.nodes[0].listwallets()), {"w1", "w2", "w3"})
+        assert_equal(set(self.nodes[0].listwallets()['loaded']), {"w1.dat", "w2.dat", "w3.dat"})
 
         self.stop_node(0)
 
         # should not initialize if there are duplicate wallets
-        self.assert_start_raises_init_error(0, ['-wallet=w1', '-wallet=w1'], 'Error loading wallet w1. Duplicate -wallet filename specified.')
+        self.assert_start_raises_init_error(0, ['-wallet=w1.dat', '-wallet=w1.dat'], 'Error loading wallet w1.dat. Duplicate -wallet filename specified.')
 
         # should not initialize if wallet file is a directory
-        os.mkdir(os.path.join(self.options.tmpdir, 'node0', 'regtest', 'w11'))
-        self.assert_start_raises_init_error(0, ['-wallet=w11'], 'Error loading wallet w11. -wallet filename must be a regular file.')
+        os.mkdir(os.path.join(self.options.tmpdir, 'node0', 'regtest', 'w11.dat'))
+        self.assert_start_raises_init_error(0, ['-wallet=w11.dat'], 'Error loading wallet w11.dat. -wallet filename must be a regular file.')
 
         # should not initialize if wallet file is a symlink
-        os.symlink(os.path.join(self.options.tmpdir, 'node0', 'regtest', 'w1'), os.path.join(self.options.tmpdir, 'node0', 'regtest', 'w12'))
-        self.assert_start_raises_init_error(0, ['-wallet=w12'], 'Error loading wallet w12. -wallet filename must be a regular file.')
+        os.symlink(os.path.join(self.options.tmpdir, 'node0', 'regtest', 'w1.dat'), os.path.join(self.options.tmpdir, 'node0', 'regtest', 'w12.dat'))
+        self.assert_start_raises_init_error(0, ['-wallet=w12.dat'], 'Error loading wallet w12.dat. -wallet filename must be a regular file.')
+
+        # should not initialize if the specified walletdir does not exist
+        self.assert_start_raises_init_error(0, ['-walletdir=bad'], 'Error: Specified wallet directory "bad" does not exist.')
+
+        # running the node with specified walletdir should only have the default wallet in it
+        os.mkdir(os.path.join(self.options.tmpdir, 'node0', 'regtest', 'walletdir'))
+        self.start_node(0, ['-walletdir=' + os.path.join(self.options.tmpdir, 'node0', 'regtest', 'walletdir')])
+        assert_equal(set(self.nodes[0].listwallets()['loaded']), {"wallet.dat"})
```

Comment:
This test should also test having multiple wallets in the walletdir, and issuing an RPC command to one of them.